### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by Christelle-12**
## Hello in this PR I 

- [ ] optmised database performance with n+1 eager solution
Screenshot from 2023-08-01 14-52-15

image
- [ ] Used Capybara to write integration tests for each view in my project.
- [ ] Tested the  User index page with these scenarios:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.

- [ ] Tested user show page with these scenarios :
I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.

- [ ] Tested User post index page with :
I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.

- [ ] Tested Post show page with:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

### CODE REVIEW

File: spec/views/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    end
+  end
+
+  scenario 'I can see the number of posts each user has written' do
+    visit users_path
+    User.all.each do |user|
+      expect(page).to have_content("Number of posts: #{user.posts_counter}")
+    end
+  end
+end
```
Comment (DammyShittu) [inline spec/views/user_index_spec.rb]: - [ ] You are missing the test that redirects to the user's show page when you click on a user.

File: spec/views/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    scenario 'When I click a users post, it redirects me to that posts show page.' do
+      visit user_path
+      click_link 'Post'
+      expect(page).to have_content(post.title)
+    end
+    scenario 'I can see the users first 3 posts'
+    visit user_path
+    expect(page).to have_content(user.posts.first(3))
+  end
+end
```
Comment (DammyShittu) [inline spec/views/user_show_spec.rb]: - [ ] You are missing tests for these assertions.

  - Tests that asserts you can see the user's bio.
  - Tests that asserts you can see a button that lets me view all of a user's posts.
  - Tests that redirects to the user's post's index page when you click on a see all posts.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Christelle-12 and @ridabensalem,

## Highlights
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @Christelle-12, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] All tests are passing ✔️ 
Screenshot 2023-08-02 at 19 25 30

- [ ] Good commit history ✔️ 
- [ ] All requested changes are implemented ✔️ 
- [ ] No linters errors ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**